tools/sdmpredictors/sdmpredictors_list_layers.xml

```diff
@@ -0,0 +1,200 @@
+                               label="Years Ago" help="Years for which you want the paleo layer, e.g. 6000, 21000.When 0 all years are listed."/>
+                   </section>
+                </when>
+            </conditional>
+        </repeat>
+    </inputs>
+    <outputs>
+        <collection name="datasets_output" type="list"
+            label="Datasets list">
+        <discover_datasets pattern="__name_and_ext__" />
+        </collection>
+    </outputs>
+    <tests>
+        <test>
+            <repeat name="list_layers">
+                <conditional name="choose_layer">
+                    <param name="input_layer" value="layers_paleo"/>
+                    <section name="environement">
+                        <param name="terrestrial" value="false"/>
+                        <param name="marine" value="true"/>
+                        <param name="freshwater" value="false"/>
+                    </section>
+                    <section name="advanced_options">
+                        <param name="monthly" value="true"/>
+                        <param name="version" value="0"/>
+                        <param name="model_name" value="All"/>
+                        <param name="epoch" value="All"/>
+                        <param name="years_ago" value="0"/>
+                    </section>
+                </conditional>
+            </repeat>
+            <repeat name="list_layers">
+                <conditional name="choose_layer">
+                    <param name="input_layer" value="layers_future"/>
+                    <section name="environement">
+                        <param name="terrestrial" value="false"/>
+                        <param name="marine" value="false"/>
+                        <param name="freshwater" value="true"/>
+                    </section>
+                    <section name="advanced_options">
+                        <param name="monthly" value="true"/>
+                        <param name="version" value="0"/>
+                        <param name="scenario" value="All"/>
+                        <param name="year" value="0"/>
+                    </section>
+                </conditional>
+            </repeat>
+            <repeat name="list_layers">
+                <conditional name="choose_layer">
+                    <param name="input_layer" value="layers_current"/>
+                    <section name="environement">
+                        <param name="terrestrial" value="true"/>
+                        <param name="marine" value="false"/>
+                        <param name="freshwater" value="false"/>
+                    </section>
+                    <section name="advanced_options">
+                        <param name="monthly" value="true"/>
+                        <param name="version" value="0"/>
+                    </section>
+                </conditional>
+            </repeat>
+            <output_collection name="datasets_output" type="list" count="3"/>
```

Comment:
You could add here asserts to test the outputs.

Reply:
You might also want to test for the correct datatyp here. E.g. with ftype=

Reply:
Many thanks for your review! I added the changes you recommended. I do not exactly know witch tests would be pertinent to implement for a galaxy tool, by all mean don't hesitate to mention if you think somenthing is missing.